processor/src/lib.rs

```diff
@@ -93,11 +95,17 @@ pub struct ChipletsTrace {
 
 /// Returns an execution trace resulting from executing the provided program against the provided
 /// inputs.
-pub fn execute(
+pub fn execute<ADV, STK, V>(
     program: &Program,
-    inputs: &ProgramInputs,
-) -> Result<ExecutionTrace, ExecutionError> {
-    let mut process = Process::new(program.kernel(), inputs.clone());
+    advice: ADV,
+    stack_init: STK,
+) -> Result<ExecutionTrace, ExecutionError>
```

Comment:
I would prefer not to change external interfaces in this PR - especially since we'll need to change them again in the next PR. So, for now, we can probably just instantiate `BaseAdviceProvider` in this function.

Reply:
Done!